python_modules/libraries/dagster-deltalake/dagster_deltalake/handler.py

```diff
@@ -55,28 +55,56 @@ def handle_output(
         connection: TableConnection,
     ):
         """Stores pyarrow types in Delta table."""
+        assert context.metadata is not None
+        assert context.resource_config is not None
```

Comment:
`assert`s get erased in optimized bytecode. If we want to raise an error, an if statement is more appropriate, in case it's just about type checking, there is a function `cast` in the `typing` module.

Alternative you could do something like

```py
metadata = context.metadata or {}
```

Reply:
Yeah it was just for type checking!